### ISSUE TITLE
Sprint 1.32.0 1/2 - Dec 13 to Jan 7

### ISSUE THREAD

**Issue by marcushyett-ph** [PostHog/posthog#7526]
# Global Sprint Planning

* This sprint will be a 4 week sprint (so folks can hit the ground running after holidays - mid two weeks will be a hackathon for those around)
* [What does an ambitious goal look like](https://posthog.com/handbook/engineering/development-process#what-does-an-awesome-big-ambitious-goal-look-like) - to help with planning

## Retro: Status of Outcomes from Previous Sprint

Owners please update with status: (**Nailed it**, **Scraped it**, **Failed it**) - if Scraped it, ensure you give enough capacity to finish off next sprint.

1. **Groups wrapup @macobo** Mostly ready, delayed by performance optimizations. Upsell live but needs iteration, person modal would be great to have, docs making good progress, tracking TBD. Might need to delay the code freeze <2 days to wrap up. 
1. **Actor modal @EDsCODE** (Add Status Update here please)
2. **Defining experiments with a clear goal (e.g. conversion, retention, volume, ...) and three clear outcomes (success, failure, inconclusive) that can be quickly identified. @neilkakkar** -> Scraped It
    - Cut down scope based on new research. Only supporting conversion experiments.
    - Won't release to test users by end of this sprint - early next week
    - Proper working experiment flow should be up by end of sprint!
3. **"make insights and dashboards not a frustrating experience anymore" @paolodamico** -> **Scraped it 🟠** _(most important themes were addressed)_
4. **"perfect playback experience for session recordings" @rcmarron**  -> The multi-window player is on track to be done at end of the sprint.
5. **Release system to do events table migration @yakkomajuri**  -> We have a full system ready, including a management UI and a good chunk of tests. However, the key piece of it all remains unreviewed as of right now https://github.com/PostHog/posthog/pull/7446)
6. **Upgrade Helm chart to k8s 1.22 @guidoiaquinti**  (Add Status Update here please) - **Nailed it**

## Retro: What can we do better next sprint?
Please edit this ticket and add any retro items in advance

1. @neilkakkar : Owners ought to be decided a sprint before we actually start implementing. This allows for background thought processes on what to do / how to do things. This sprint felt too immediate: I became an owner on Monday, which meant 2-3 days were spent catching up on what the world knows about a/b testing, and coming up with a reasonable approach. Like in correlation, a lot of this can happen in the background during previous sprints, which helps us hit the ground running.
2. Core-Exp: Bit off more than we could chew, prioritized by most important first. Dropping some stuff for later
3. Core-Analytics:Upsell work on groups was without a clear owner until EOW last week which added quite a bit of confusion. :) Side-effect of product people being in two teams: Action - clarify ownership timely

## Plan: Proposed Goals for Next Sprint 

Each goal should have a single owner. Owner can only be an engineer.

1. **Plugin Server Scalability:** Fix overloading issues on plugin server worker threads @yakkomajuri 
2. **Nail Deployments:** CLI to make helm chart deploys seamless @fuziontech 
3. **Performance:** Optimize most query performance large customers to <5seconds @macobo 
4. **Experimentation:** Users can run multiple multivariate experiments & take decisions based on the results of those experiments @neilkakkar 
5. **Nail Insights date support** @pauldambra 
6. **Make Dashboards great again** @Twixes 
7.  **Collaboration: Taxonomy:** Users should understand events actions and properties others have set up @alexkim205 

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
Team ___

## Retro

<!-- Talk about what went well, what didn't go well and any actions to improve next time -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## Planning

<!-- Each item should have a single owner. Owner can only be an engineer. -->

### High priority

-

### Low priority / side quests

-

```

**Comment by yakkomajuri** [comment]
# Team Platform

## Retro

### Goals

- Release system to do events table migration @yakkomajuri -> We have a full system ready, including a management UI and a good chunk of tests. However, the key piece of it all remains unreviewed as of right now Special Migrations Runner (#7054 pt. 4) #7446)
- Upgrade Helm chart to k8s 1.22 @guidoiaquinti (Add Status Update here please) - Nailed it

### What could be better

- Support hero takes more than 8h/day 
- 2/4 team members were support heroes (+ some were on vacation)
- Potentially worth rethinking support hero (not scalable)

## Hang over items from previous sprint

- Last mile on #6834 @tiina303: reprioritize

## Planning

### High priority

- Upgrading ClickHouse @fuziontech 
- Plugin Server Scalability: Fix overloading issues on plugin server worker threads @yakkomajuri
- Nail Deployments: CLI to make helm chart deploys seamless @guidoiaquinti (+ @fuziontech)
- Last mile on #6834 @tiina303 

### Low priority / side quests

https://github.com/orgs/PostHog/projects/10

**Comment by EDsCODE** [comment]
Team Core Analytics

## Retro



- Harry: Consolidating retention UI and queries, J curve PR open. Got tied up in E2E test issues
- Eric: Actor work mostly complete. Got slowed down by various investigations and trying to keep up with Phantom experience
- Li: Experimentation frontend visualizations. Snagged on antd 
- Neil: Backend for experiementation
- Karl: ???

## Hang over items from previous sprint



- [Experiment MVP finish](https://github.com/PostHog/posthog/issues/7462): prioritize
- Insight csvs: prioritize during support week if possible

## Planning



### High priority

**Performance improvement: phantom (large client with >10M users) can run _most_ queries over 30 days in <5s — Owner: Karl**
- Retention https://github.com/PostHog/posthog/issues/7595
- Person related queries https://github.com/PostHog/posthog/issues/7548
- Lifecycle https://github.com/PostHog/posthog/issues/7382
- Slow breakdown queries on Phantom
- (Making sure funnel, correlations, and paths are performant for phantom)

**Experimentation: Users can run multiple multivariate experiments & take decisions based on the results of those experiments — Owner: Neil**
- Gather feedback
     - Are users receptive to bayesian format?
- Support multivariates (creation flow + results)
- Provide more data to make decisions - (1) [Histograms of effect size](https://github.com/PostHog/posthog/issues/7418#issuecomment-984616174). (2) <Whatever users say they want>

**Comment by mariusandra** [comment]
Team Core Experience

## Retro

- Rick: work was a bit weird -> refactoring something that was just built (recordings player). Should we have started with the big goal (multiple windows) from the get go?
- Paolo: we're too optimistic (tooltips, taxonomy), so now we need to rearrange things. Let's be more conservative on Wednesdays. Is pairing within a sprint good? Should we do more/less? Pairing makes sense.
- Michael: okayish. Would've preferred to focus on the big stuff (dashboards), distracted by smaller stuff. Been a bit tough to work with Chris on designs due to timezones (ping-pong). 
- Chris: agree with Michael, sorting out what's going to happen next, more balancing between teams upcoming. Clarify expectations on working with designers.
- Paul: 2 weeks ago is horribly long ago... two 4 day weeks, first really productive, last week was a week of sickness. Good: people support when family is ill. Pairing was good, to know I can focus on some reviews over others.
- Alex: Didn't get to focus on a lot of coding, overpromising. Bogged down by long PRs (mine) to review. One outstanding for 10 days, got pulled in many small directions and did what I could to unblock people. Hard to move in/out of context. Be more realistic on what can get done.
- Marius: plenty of sick time (myself and family), secondary rotation for core experience took a bit of time, got some stuff done, but not all priorities.

## Headcount

- Marius: 1 week here (sick family), 1 week support hero, then 2 weeks on holiday
- Chris: 1 week here, off 2 weeks, 1 week back
- Paolo: 1 week here, off 2 weeks, 1 week back
- Alex: 1 week here, off 2 weeks, 1 week back
- Rick: 1.5 week here, off periodically after, working ~1.5 more weeks
- Paul: 3 day weeks each of the next 4 weeks
- Michael: 1.6 weeks now, out for 1.4 weeks, then back for 1 week

## Planning

https://github.com/PostHog/posthog/issues/7581

**Comment by yakkomajuri** [comment]
@posthog-bot /remind define #platform as @tiina303 @fuziontech @guidoiaquinti @yakkomajuri

**Comment by yakkomajuri** [comment]
@posthog-bot /remind define #platform as @tiina303 @fuziontech @guidoiaquinti @yakkomajuri

**Comment by yakkomajuri** [comment]
@posthog-bot /remind define #team-platform as @tiina303 @fuziontech @guidoiaquinti @yakkomajuri

**Comment by yakkomajuri** [comment]
@posthog-bot /remind #team-platform to do the sprint retro on 07/01/2021 at 2pm UTC

**Comment by yakkomajuri** [comment]
@posthog-bot /remind #team-platform to do the sprint retro on 07/01/2021 at 2pm UTC

**Comment by yakkomajuri** [comment]
@posthog-bot /remind #team-platform to do the sprint retro on 01/07/2021 at 2pm UTC

**Comment by yakkomajuri** [comment]
@posthog-bot /remind #team-platform to do the sprint retro on 01/07/2022 at 2pm UTC

**Comment by yakkomajuri** [comment]
@posthog-bot /remind list

**Comment by yakkomajuri** [comment]
@posthog-bot /remind delete 9

**Comment by yakkomajuri** [comment]
@posthog-bot /remind help
